src/Microsoft.DotNet.Tools.Builder/CompileContext.cs

```diff
@@ -0,0 +1,248 @@
+            foreach (var dependency in Sort(_dependencies))
+            {
+                if (!InvokeCompileOnDependency(dependency, _outputPath, _intermediateOutputPath))
+                    return false;
+            }
+
+            //compile project
+            var success = InvokeCompileOnRootProject(_outputPath, _intermediateOutputPath);
+
+            PrintSummary(success);
+
+            return success;
+        }
+
+        private void PrintSummary(bool success)
+        {
+            if (success)
+            {
+                Reporter.Output.Write(" " + _preconditions.LogMessage());
+                Reporter.Output.WriteLine();
+            }
+
+            Reporter.Output.WriteLine();
+        }
+
+        private void CreateOutputDirectories()
+        {
+            Directory.CreateDirectory(_outputPath);
+            Directory.CreateDirectory(_intermediateOutputPath);
+        }
+
+        //todo make extension of ProjectContext?
+        //returns map with dependencies: string projectName -> ProjectDescription
+        private static Dictionary<string, ProjectDescription> GetProjectDependenciesWithSources(ProjectContext projectContext, string configuration)
+        {
+            var projects = new Dictionary<string, ProjectDescription>();
+
+            // Create the library exporter
+            var exporter = projectContext.CreateExporter(configuration);
+
+            // Gather exports for the project
+            var dependencies = exporter.GetDependencies().ToList();
+
+            // Build project references
+            foreach (var dependency in dependencies)
+            {
+                var projectDependency = dependency.Library as ProjectDescription;
+
+                if (projectDependency != null && projectDependency.Project.Files.SourceFiles.Any())
+                {
+                    projects[projectDependency.Identity.Name] = projectDependency;
+                }
+            }
+
+            return projects;
+        }
+
+        private IncrementalPreconditions GatherIncrementalPreconditions()
+        {
+            var preconditions = new IncrementalPreconditions(_args.BuildProfileValue);
+
+            var projectsToCheck = GetProjectsToCheck();
+
+            foreach (var project in projectsToCheck)
+            {
+                CollectScriptPreconditions(project, preconditions);
+                CollectCompilerNamePreconditions(project, preconditions);
+                CheckPATHProbing(project, preconditions);
+            }
+
+            return preconditions;
+        }
+
+        //check the entire project tree that needs to be compiled, duplicated for each framework
+        private List<ProjectContext> GetProjectsToCheck()
+        {
+            //include initial root project
+            var contextsToCheck = new List<ProjectContext> {_rootProject};
+
+            //convert ProjectDescription to ProjectContext
+            var dependencyContexts = _dependencies.Select
```

Comment:
So it should be
`ProjectContext.Create(keyValuePair.Value.Path, keyValuePair.Value.Framework))`
instead of the current
`ProjectContext.Create(keyValuePair.Value.Project.ProjectFilePath, _rootProject.TargetFramework))`
?